src/main/java/seedu/address/logic/commands/QueryDoctorCommand.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.person.DoctorNameContainsKeywordsPredicate;
+
+/**
+ * Queries and returns all doctors whose name matches the input string.
+ * Keyword matching is case insensitive.
+ * Query more than one name at a time is supported
+ */
+public class QueryDoctorCommand extends Command {
+
+    public static final String COMMAND_WORD = "querydoctor";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all doctors whose names contain any of "
+            + "the specified keywords (case-insensitive) and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " alice bob charlie";
+
+    private final DoctorNameContainsKeywordsPredicate predicate;
```

Comment:
Perhaps using a variable name like doctorNamePredicate would be more descriptive/clearer than just a predicate? Just something to think about.